var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -58,3 +59,33 @@ def cmake_args(self):
         ]
 
         return args
+
+    @run_after("install")
+    def setup_build_tests(self):
+        launcher = self.mpi_launcher()
+
+        filter_file(r"-n 2", "-n 2 --timeout 240", "samples/sync-write-1d-contig-contig.json")
```

Comment:
So what you want is something along the lines:

```suggestion
        exe = os.path.basename(launcher.path)
        filename = "samples/sync-write-1d-contig-contig.json"
        filter_file(f"{exe}", f"{launcher}", filename)
        filter_file(r"-n 2", "-n 2 --timeout 240", filename)
```

Kudos for the addition of the timeout as it will be useful for CI since most containers won't be able to run say `mpirun`.  

If it does time out then whomever is checking the test results will have to know to look at `stderr` in the appropriate subdirectory of the test stage (whose root directory is included in the note to see the <pkg>-test-out.txt file.

Note: I didn't track but at one point I was trying to run this with a launcher that doesn't support `--timeout` so that could be a problem.

Reply:
@tldahlgren hmm, should we approach this in a different way because of the `--timeout` support?

Reply:
How so?  Do you have tests that could be run with a single process?

We have discussed several ways to add timeout support in the Spack core but those I've investigated so far haven't had the desired affect.

Reply:
No, all useful tests require MPI. At this point, maybe we should remove the timeout, as it could cause failures for some MPI implementations. Should we keep it or remove it?

Reply:
You can keep it. There are other ways we can deal with the package in CI.

Reply:
Quick question though, since I'd rather not disable the tests for CI if not necessary:  Will the test still work if the number of processes is set to 1?

Reply:
Yes, it should work with 1 process, I've pushed a change for that